crates/reth/evm/src/precompiles/schnorr.rs

```diff
@@ -0,0 +1,114 @@
+use revm::{
+    precompile::utilities::right_pad,
+    primitives::{PrecompileOutput, PrecompileResult},
+};
+use revm_primitives::Bytes;
+use strata_crypto::verify_schnorr_sig;
+use strata_primitives::buf::{Buf32, Buf64};
+
+/// Internal representation of parsed Schnorr input bytes.
+struct SchnorrInput {
+    /// 32 Bytes: Public key
+    public_key: Buf32,
+    /// 32 Bytes: Message hash
+    message_hash: Buf32,
+    /// 64 Bytes: Schnorr Signature
+    signature: Buf64,
+}
+
+fn parse_schnorr_input(input: &Bytes) -> SchnorrInput {
+    let input = right_pad::<128>(input);
+
+    SchnorrInput {
+        public_key: Buf32::new(input[0..32].try_into().unwrap()),
+        message_hash: Buf32::new(input[32..64].try_into().unwrap()),
+        signature: Buf64::new(input[64..128].try_into().unwrap()),
+    }
+}
+
+pub fn verify_schnorr_precompile(input: &Bytes, _gas_limit: u64) -> PrecompileResult {
+    let schnorr_input = parse_schnorr_input(input);
+
+    let result = verify_schnorr_sig(
+        &schnorr_input.signature,
+        &schnorr_input.message_hash,
+        &schnorr_input.public_key,
+    );
+    let verification_byte = Bytes::from([result as u8]);
+
+    // currently we can use [ecrecover hack](https://hackmd.io/@nZ-twauPRISEa6G9zg3XRw/SyjJzSLt9)
+    // which costs around ~3000 gas.
+    // setting it as 0, as this requires further discussion
+    let gas_cost = 0;
```

Comment:
Created ticket [STR-864](https://alpenlabs.atlassian.net/browse/STR-864) for this.